File: clients/lumify-sdk-python/lumify/resources/teams.py
from __future__ import annotations

from typing import Any, Iterator, Optional

from .._transport import LumifyClient
from ..models import Team, TeamsListResponse
from ..pagination import iterate_items, paginate


class TeamsResource:
    def __init__(self, client: LumifyClient) -> None:
        self._client = client

    def list(
        self,
        *,
        sport: Optional[str] = None,
        league: Optional[str] = None,
        conference: Optional[str] = None,
        division: Optional[str] = None,
        country: Optional[str] = None,
        q: Optional[str] = None,
        active: Optional[bool] = None,
        after_id: Optional[int] = None,
        limit: Optional[int] = None,
    ) -> TeamsListResponse:
        """GET /v1/teams — cursor-paginated."""
        return self._client.get(
            "/v1/teams",
            query={
                "sport": sport,
                "league": league,
                "conference": conference,
                "division": division,
                "country": country,
                "q": q,
                "active": active,
                "after_id": after_id,
                "limit": limit,
            },
        )

    def paginate(
        self, *, limit: int = 25, max_pages: int = 1000, **filters: Any
    ) -> Iterator[TeamsListResponse]:
        return paginate(
            lambda after_id, lim: self.list(after_id=after_id, limit=lim, **filters),
            limit=limit,
            max_pages=max_pages,
        )

    def iterate(
        self, *, limit: int = 25, max_pages: int = 1000, **filters: Any
    ) -> Iterator[Any]:
        return iterate_items(
            lambda after_id, lim: self.list(after_id=after_id, limit=lim, **filters),
            items_key="data",
            limit=limit,
            max_pages=max_pages,
        )

    def get(self, team_id: int) -> Team:
        """GET /v1/teams/{id}."""
        return self._client.get("/v1/teams/%d" % team_id)

File: clients/lumify-sdk/src/resources/teams.ts
import type { LumifyClient } from "../client.js";
import type { Team, TeamsListResponse } from "../generated/models.js";
import { paginate, iterateItems, type PaginateOptions } from "../pagination.js";

export interface ListTeamsParams {
  sport?: string;
  league?: string;
  conference?: string;
  division?: string;
  country?: string;
  /** Free-text search on team name/abbreviation. */
  q?: string;
  active?: boolean;
  afterId?: number;
  /** Max 100, default 25. */
  limit?: number;
}

function toQuery(p: ListTeamsParams) {
  return {
    sport: p.sport,
    league: p.league,
    conference: p.conference,
    division: p.division,
    country: p.country,
    q: p.q,
    active: p.active,
    after_id: p.afterId,
    limit: p.limit,
  };
}

export class TeamsResource {
  constructor(private readonly client: LumifyClient) {}

  /** GET /v1/teams — cursor-paginated. */
  list(params: ListTeamsParams = {}): Promise<TeamsListResponse> {
    return this.client.get<TeamsListResponse>("/v1/teams", { query: toQuery(params) });
  }

  /** Async-iterate every page of `list()` for the given filters. */
  paginate(params: Omit<ListTeamsParams, "afterId"> = {}, options: PaginateOptions = {}) {
    return paginate(
      (afterId, limit) => this.list({ ...params, afterId, limit }),
      { limit: params.limit, ...options }
    );
  }

  /** Async-iterate every team matching the filters, across all pages. */
  iterate(params: Omit<ListTeamsParams, "afterId"> = {}, options: PaginateOptions = {}) {
    return iterateItems(
      (afterId, limit) => this.list({ ...params, afterId, limit }),
      { limit: params.limit, ...options }
    );
  }

  /** GET /v1/teams/{id} */
  get(teamId: number): Promise<Team> {
    return this.client.get<Team>(`/v1/teams/${teamId}`);
  }
}
